synapse/rest/media/v1/preview_html.py

```diff
@@ -321,14 +321,33 @@ def _iterate_over_text(
 
 
 def rebase_url(url: str, base: str) -> str:
-    base_parts = list(urlparse.urlparse(base))
+    """
+    Resolves a potentially relative `url` against an absolute `base` URL.
```

Comment:
Do you know how this differs from [`urljoin`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin) in the standard lib?
Not necessarily opposed to rolling our own, but it'd be nice to know if we had specific motivation for doing so.

Reply:
I didn't know of it, but I'll look into it!

Reply:
I *think* this will work OK, but I'd rather punt this to a separate PR since it is pretty unrelated to this work. Would that be OK?

Reply:
yeah OK